lib/ace/edit_session.js

```diff
@@ -2398,9 +2399,10 @@ EditSession.$uid = 0;
     };
     
     this.destroy = function() {
-        if (this.bgTokenizer) {
+        if (!this.destroyed) {
             this.bgTokenizer.setDocument(null);
-            this.bgTokenizer = null;
```

Comment:
The main reason to set bgTokenizer to null was to clean states and lines arrays on it, in case session is retained due to memory leak.

It may be better to add bgTokenizer destroy method that would remove these and call other bgTokenizer cleanup methods.

Reply:
okay, makes sense, I’ll add destroy method

Reply:
@nightwing updated my PR with `cleanup` method